tests/theseus_tests/core/test_objective.py

```diff
@@ -547,3 +547,18 @@ def test_to_dtype():
                 assert var.dtype == dtype
             for aux in cf.aux_vars:
                 assert aux.dtype == dtype
+
+def test_cost_delete_and_add():
+    x = th.Variable(torch.zeros(1), name="x")
+    y = th.Variable(torch.zeros(1), name="y")
+
+    def error_fn(optim_vars, aux_vars):
+        return 0
+
+    objective = th.Objective()
+    cost_function = th.AutoDiffCostFunction([x], error_fn, 1, aux_vars=[y], cost_weight=th.ScaleCostWeight(1.0))
+    
+    # Add a cost function, erase it, and add it again to make sure we don't have any bugs.
+    objective.add(cost_function)
+    objective.erase(cost_function.name)
+    objective.add(cost_function)
```

Comment:
Thanks for adding the unit test. Might need some asserts to check if it does the right thing?

Any thoughts @luisenp?

Reply:
Test failure is caused by throwing an exception. If the last statement doesn't throw an exception, the test passes.

Reply:
We could still check if `objective.optim_vars`, `objective.aux_vars`, and `objective.cost_functions` returns the right thing after adding, erasing and adding again.

Reply:
@Hesam-Vayu let me know if you have any questions about the suggestions above. Thanks!

Reply:
I added some checks. I used the 'x.name in optim_vars' pattern as there's no guarantee that other variables will not be created during add cost function operations and enforcing the constraint that only passed in variables should be added in this test would severely restrict the design space of the system.

Reply:
@luisenp Any thoughts on the tests?